src/ir/module-utils.h

```diff
@@ -128,6 +129,52 @@ inline void copyModule(Module& in, Module& out) {
   out.debugInfoFileNames = in.debugInfoFileNames;
 }
 
+// Renaming
+
+template<typename T>
+inline void renameFunctions(Module& wasm, T& map) {
+  // Update the function itself.
+  for (auto& pair : map) {
+    assert(!wasm.getFunctionOrNull(pair.second));
+    wasm.getFunction(pair.first)->name = pair.second;
+  }
+  wasm.updateMaps();
+  // Update other global things.
+  auto maybeUpdate = [&](Name& name) {
+    auto iter = map.find(name);
+    if (iter != map.end()) {
+      name = iter->second;
+    }
+  };
+  maybeUpdate(wasm.start);
+  for (auto& segment : wasm.table.segments) {
+    for (auto& name : segment.data) {
+      maybeUpdate(name);
```

Comment:
This part is confusing me.  Why would segments get renamed by renameFunctions()?

I can see the previous code did this too but I can't see why ...